### PR TITLE
allow converting from/to TwistWithCovarianceStamped

### PR DESCRIPTION
**Pull request by ejalaa12**
# 🎉 New feature

Allows converting from/to TwistWithCovarianceStamped

## Summary

`ros_gz_bridge` allows converting `gz.msgs.TwistWithCovariance` <-> `geometry_msgs.msg.TwistWithCovariance`.

However the `gz.msgs.TwistWithCovariance` also contains a header and stamp. So it could be converted to a `geometry_msgs.msg.TwistWithCovariance**Stamped**`.

This PR implements that conversion. It is now possible to convert either to/from:
- `geometry_msgs.msg.TwistWithCovariance`
- `geometry_msgs.msg.TwistWithCovarianceStamped`.

Note: I also have a backport of that commit to the humble branch, should I open it after this PR is done ?

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ros_gz_bridge/README.md [hunk 1]
```diff
@@ -5,61 +5,61 @@ between ROS and Gazebo Transport.
...
+| ros_gz_interfaces/msg/Contacts               | ignition::msgs::Contacts               |
+| ros_gz_interfaces/msg/Dataframe              | ignition::msgs::Dataframe            |
+| ros_gz_interfaces/msg/Entity                 | ignition::msgs::Entity                 |
+| ros_gz_interfaces/msg/Float32Array           | ignition::msgs::Float_V                |
+| ros_gz_interfaces/msg/GuiCamera              | ignition::msgs::GUICamera              |
+| ros_gz_interfaces/msg/JointWrench            | ignition::msgs::JointWrench            |
+| ros_gz_interfaces/msg/Light                  | ignition::msgs::Light                  |
+| ros_gz_interfaces/msg/StringVec              | ignition::msgs::StringMsg_V            |
+| ros_gz_interfaces/msg/TrackVisual            | ignition::msgs::TrackVisual            |
+| ros_gz_interfaces/msg/VideoRecord            | ignition::msgs::VideoRecord            |
```
Comment (adityapande-1995) [inline ros_gz_bridge/README.md]: Could you fix the vertical table alignment and keep the namespace as "gz::" ?

**Review by adityapande-1995 — CHANGES_REQUESTED** [review]
Thanks for adding this ! 
I can see we can add PoseWithCovarianceStamped in a similar fashion. Though not required in this PR, could you add a ticket for this ?

File: ros_gz_bridge/README.md (hunk 1 above)
Comment (ejalaa12) [reply ros_gz_bridge/README.md]: Nice catch, I was probably working on a wrong branch with ignition instead of gz. I'll a an amend commit to clean this up

**Comment by ejalaa12** [comment]
sorry for the delay in answer @adityapande-1995 
I just pushed the changes for the readme.
I also opened up a PR for the PoseWithCovarianceStamped as well.

**Comment by ejalaa12** [comment]
@adityapande-1995 Did you have time to review my changes ?

**Review by ahcorde — CHANGES_REQUESTED** [review]
conflicts

**Comment by ejalaa12** [comment]
@ahcorde just fixed the conflicts

**Review by ahcorde — APPROVED** [review]
@adityapande-1995 do you mind to take another look ?

**Review by adityapande-1995 — APPROVED** [review]
Looks good with conflicts resolved in the README !

**Comment by ejalaa12** [comment]
Thanks for the review and the merge !

**Pull request merged**